src/compiler/checker.ts

```diff
@@ -37085,7 +37085,12 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function checkVoidExpression(node: VoidExpression): Type {
-        checkExpression(node.expression);
+        if (node.parent.kind === SyntaxKind.ArrowFunction && (node.parent as ArrowFunction).body === node) {
+            checkNodeDeferred(node.expression);

```

Comment:
After a second thought... perhaps it's just always OK to defer those expressions? I don't see what could go wrong, regardless of the position in the file :p

Reply:
"Always defer" seems reasonable. The code as written seems to just hardcode the reported repro syntax tree, which is not really a good sign